wicket-core/src/main/java/org/apache/wicket/IMetadataContext.java

```diff
@@ -0,0 +1,10 @@
+package org.apache.wicket;
+
+/**
+ * @author Jezza
+ */
+public interface IMetadataContext<B, R> {
```

Comment:
I don't like the names of the generics, nor the R parameter.
It was only added because the current implementation returns the object the method is enclosed in, so right now, it just conveys that type information.  
Ideally, I would want to remove it.

One solution would be just to return the IMetadataContext<B> itself, but you wouldn't be able to do much with the return result, short of just setting more metadata.

Reply:
I also have doubts about the interface name.
If anyone has a better suggestion, I'd be happy to swap it out.